### PR TITLE
Development config

### PR DESCRIPTION
**Pull request by marcelveldt**
a few changes for development with VS Code on MacOS

### CODE REVIEW

**Comment by marcelveldt** [comment]
This ESLint config is more strict. What to do ? Adjust code so the linting works reliable or loosen it ?

File: .eslintrc.js [hunk 1]
```diff
@@ -10,4 +10,7 @@ module.exports = {
     sourceType: "module",
   },
   plugins: ["@typescript-eslint"],
+  rules: {
+    "no-unused-vars": "off",
```
Comment (robertsLando) [inline .eslintrc.js:14]: @marcelveldt My suggestion is to keep this on and add the ignore lines where needed. This rule helps

**Review by robertsLando — CHANGES_REQUESTED** [review]

**Comment by marcelveldt** [comment]
We should address all warnings that the linter has found with these stricter linting config.
I'll do that when I get bored next couple of days ;-)

File: src/lib/error.ts [hunk 2]
```diff
@@ -1,7 +1,8 @@
+/* eslint-disable no-unused-vars */
```
Comment (robertsLando) [inline src/lib/error.ts:1]: Based on https://stackoverflow.com/questions/57802057/eslint-configuring-no-unused-vars-for-typescript

YOu could remove all this lines, just try to update `typescript-eslint/recommended` to latest version

**Review by robertsLando — CHANGES_REQUESTED** [review]

File: src/lib/error.ts (hunk 2 above)
Comment (marcelveldt) [reply src/lib/error.ts:1]: Yeah, I just figured out this is fixed in the latest version

Comment (marcelveldt) [reply src/lib/error.ts:1]: @robertsLando one more question. It's now forcing a tabwidth of 2 spaces, is that correct?

**Comment by marcelveldt** [comment]
Why is CI still testing with the old config? All tests run fine on my machine. Looks as if the CI task is using some cached info or is not using the linter config from this PR.

**Comment by marcelveldt** [comment]
This one is finally done. It's using the recommended setup of ESlint with the prettier plugin and typescript support now.

File: src/lib/error.ts (hunk 2 above)
Comment (robertsLando) [reply src/lib/error.ts]: Dunno if it's correct or not I mean it's a matter of personal choice, you can choose the preferred indentation

Comment (marcelveldt) [reply src/lib/error.ts]: Yeah, I've set it back to 2. 
Kind of used to 4 in the python world ;-)

File: .eslintrc.js [hunk 3]
```diff
@@ -1,13 +1,20 @@
...
-  parserOptions: {
-    ecmaVersion: 11,
-    sourceType: "module",
+    node: true
   },
-  plugins: ["@typescript-eslint"],
+  extends: [
+    "plugin:@typescript-eslint/recommended", // Uses the recommended rules from the @typescript-eslint/eslint-plugin
+    "prettier/@typescript-eslint", // Uses eslint-config-prettier to disable ESLint rules from @typescript-eslint/eslint-plugin that would conflict with prettier
+    "plugin:prettier/recommended" // Enables eslint-plugin-prettier and eslint-config-prettier. This will display prettier errors as ESLint errors. Make sure this is always the last configuration in the extends array.
```
Comment (balloob) [inline .eslintrc.js:14]: I am against this addition. Prettier shouldn't show up as an error. We already automatically fix it before commit and people can set up format on save.

Comment (balloob) [reply .eslintrc.js:14]: It pollutes the editor when writing.

Comment (marcelveldt) [reply .eslintrc.js:14]: This change prefers ESLint with prettier plugin so it should be better for use with VS Code as it's the recommended way according to the VS Code team. What do you suggest ?

File: .prettierrc.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
```
[the remark this answers is not in the document]
Comment (marcelveldt) [reply .prettierrc.js:1]: I can leave out the prettier config so it will use the defaults ;-)

[the remark this answers is not in the document]
Comment (marcelveldt) [reply .prettierrc.js:1]: Having this file in place will make sure that no VS-Code user configured file is messing with the defaults though.

File: .eslintrc.js (hunk 3 above)
Comment (balloob) [reply .eslintrc.js:14]: It's super annoying to see ESLine errors and warnings that you know are going to be automatically formatted away when you hit save. I vote against it .

File: .prettierrc.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (balloob) [reply .prettierrc.js:1]: We have a CI check to enforce that ;) I would prefer we just go with defaults.

File: .eslintrc.js (hunk 3 above)
Comment (marcelveldt) [reply .eslintrc.js:14]: Ah, now I get what you mean now. My only reason for this PR was to enforce default formatting rules but I don't care how :-)

**Pull request closed without merging**